### PR TITLE
Refactor some more

### PR DESCRIPTION
**Pull request by Kukks**

### CODE REVIEW

File: WalletWasabi.Backend/SendPushService.cs [hunk 1]
```diff
@@ -90,15 +91,15 @@ public async Task SendNotificationsAsync(bool isDebug)
...
 				{
-					if(res.ReasonPhrase == "BadDeviceToken")
+					if (res.ReasonPhrase == "BadDeviceToken")
 					{
 						context.Tokens.Remove(token);
 					}
 				}
-			} // TODO parallelize
+			}));
+			await context.SaveChangesAsync();
```
Comment (Kukks) [inline WalletWasabi.Backend/SendPushService.cs:102]: This is actually a bug fix:  you were not saving the deletion actions

File: WalletWasabi.Backend/SendPushService.cs [hunk 2]
```diff
@@ -80,7 +81,7 @@ public async Task SendNotificationsAsync(bool isDebug)
 			var tokens = context.Tokens
 				.Where(t => t.IsDebug == isDebug)
 				.Distinct();
-			foreach (var token in tokens)
+			await Task.WhenAll(tokens.Select(async token =>
```
Comment (Kukks) [inline WalletWasabi.Backend/SendPushService.cs:84]: Parallelizing the calls

File: WalletWasabi.Backend/Global.cs [hunk 3]
```diff
@@ -90,7 +93,7 @@ public async Task InitializeAsync(Config config, CoordinatorRoundConfig roundCon
 			var indexFilePath = Path.Combine(indexBuilderServiceDir, $"Index{RpcClient.Network}.dat");
 			var blockNotifier = HostedServices.FirstOrDefault<BlockNotifier>();
 			IndexBuilderService = new IndexBuilderService(RpcClient, blockNotifier, indexFilePath);
-			Coordinator = new Coordinator(RpcClient.Network, blockNotifier, Path.Combine(DataDir, "CcjCoordinator"), RpcClient, roundConfig, () =>  pushService.SendNotificationsAsync(false));
+			Coordinator = new Coordinator(RpcClient.Network, blockNotifier, Path.Combine(DataDir, "CcjCoordinator"), RpcClient, roundConfig, () => _ = serviceProvider.GetRequiredService<SendPushService>().SendNotificationsAsync(false));
```
Comment (Kukks) [inline WalletWasabi.Backend/Global.cs:96]: Can make this cleaner by introducing an event (and  moving the coordinator into the DI)

File: WalletWasabi.Backend/Data/DesignTimeDbContextFactoryBase.cs [hunk 4]
```diff
@@ -12,37 +10,16 @@ public abstract class DesignTimeDbContextFactoryBase<TContext> :
...
-				Environment.GetEnvironmentVariable("ASPNETCORE_ENVIRONMENT"));
+			return Create();
 		}
 
 		protected abstract TContext CreateNewInstance(
 			DbContextOptions<TContext> options);
 
 		public TContext Create()
 		{
-			var environmentName =
```
Comment (Kukks) [inline WalletWasabi.Backend/Data/DesignTimeDbContextFactoryBase.cs:25]: Cleanup of all this bloat code

File: WalletWasabi.Backend/Startup.cs [hunk 5]
```diff
@@ -78,6 +78,8 @@ public void ConfigureServices(IServiceCollection services)
 
 			services.AddSingleton<IExchangeRateProvider>(new ExchangeRateProvider());
 			services.AddSingleton(new Global(Configuration["datadir"]));
+			services.AddSingleton<SendPushService>();
```
Comment (Kukks) [inline WalletWasabi.Backend/Startup.cs:81]: Starts adding stuff to DI as an introduction to cleaning up the `Global` beast

**Review by Kukks — COMMENTED** [review]
@DanGould

File: WalletWasabi.Backend/SendPushService.cs (hunk 2 above)
Comment (DanGould) [reply WalletWasabi.Backend/SendPushService.cs:84]: unfortunately the `async token => ...` lambda cannot be converted to an expression tree as argument to `tokens.Select(`

File: WalletWasabi.Backend/InitConfigStartupTask.cs [hunk 6]
```diff
@@ -57,13 +58,11 @@ public async Task ExecuteAsync(CancellationToken cancellationToken)
...
 			var cachedRpc = new CachedRpcClient(rpc, Cache);
-
-			var pushService = new SendPushService(ContextFactory, config);
-			await Global.InitializeAsync(config, roundConfig, cachedRpc, pushService, cancellationToken);
+			await Global.InitializeAsync(config, roundConfig, cachedRpc, _serviceProvider, cancellationToken);
 
 			try
 			{
-				await WebsiteTorifier.CloneAndUpdateOnionIndexHtmlAsync();
+				await _websiteTorifier.CloneAndUpdateOnionIndexHtmlAsync();
```
Comment (DanGould) [inline WalletWasabi.Backend/InitConfigStartupTask.cs:65]: Wasabi has code cleanup rules that yell at you for this private readonly naming convention

**Pull request merged**